ts/src/hashkey.ts

```diff
@@ -0,0 +1,4336 @@
+        //         ...
+        //     ]
+        //
+        return this.parseTrades (response, market, since, limit);
+    }
+
+    async fetchMyTrades (symbol: Str = undefined, since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name hashkey#fetchMyTrades
+         * @description fetch all trades made by the user
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/get-account-trade-list
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/query-futures-trades
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/get-sub-account-user
+         * @param {string} symbol *is mandatory for swap markets* unified market symbol
+         * @param {int} [since] the earliest time in ms to fetch trades for
+         * @param {int} [limit] the maximum amount of trades to fetch (default 200, max 500)
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {string} [params.type] 'spot' or 'swap' - the type of the market to fetch trades for (default 'spot')
+         * @param {int} [params.until] the latest time in ms to fetch trades for, only supports the last 30 days timeframe
+         * @param {string} [params.fromId] srarting trade id
+         * @param {string} [params.toId] ending trade id
+         * @param {string} [params.clientOrderId] *spot markets only* filter trades by orderId
+         * @param {string} [params.accountId] account id to fetch the orders from
+         * @returns {Trade[]} a list of [trade structures]{@link https://github.com/ccxt/ccxt/wiki/Manual#trade-structure}
+         */
+        const methodName = 'fetchMyTrades';
+        await this.loadMarkets ();
+        const request: Dict = {};
+        let market: Market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        let marketType = 'spot';
+        [ marketType, params ] = this.handleMarketTypeAndParams (methodName, market, params);
+        if (since !== undefined) {
+            request['startTime'] = since;
+        }
+        if (limit !== undefined) {
+            request['limit'] = limit;
+        }
+        let until: Int = undefined;
+        [ until, params ] = this.handleOptionAndParams (params, methodName, 'until');
+        if (until !== undefined) {
+            request['endTime'] = until;
+        }
+        let fromId: Str = undefined;
+        [ fromId, params ] = this.handleOptionAndParams (params, methodName, 'fromId');
+        if (fromId !== undefined) {
+            request['fromId'] = fromId;
+        }
+        let toId: Str = undefined;
+        [ toId, params ] = this.handleOptionAndParams (params, methodName, 'toId');
+        if (toId !== undefined) {
+            request['toId'] = toId;
+        }
+        let accountId: Str = undefined;
+        [ accountId, params ] = this.handleOptionAndParams (params, methodName, 'accountId');
```

Comment:
same here, we should keep them in the docs, but we don't need this.

Reply:
The same - just to search 'preinstalled' values in `options`

Reply:
but in this case it is not needed because "fromId/toId is not a "fixed" value, users have to adjust it frequently so it's not a value to store in options